fmt/src/ast_eq.rs

```diff
@@ -0,0 +1,508 @@
+use itertools::Itertools;
+use solang_parser::pt::*;
+
+pub trait AstEq {
```

Comment:
Is the reason why this trait exists because we cannot derive on the larpop code-gened files? https://github.com/hyperledger-labs/solang/blob/c63b552373f9660027568e0af4040a82d8821de7/solang-parser/src/solidity.lalrpop#L20

@seanyoung is there a way for us to do that?

Reply:
Yeah, agree. A derive macro would definitely make this more succinct, but we would have to cover the cases where order matters and cases where they don't which means making the proc_macro a little more complicated. In the end, because I just use this for writing tests to make sure the formatted parse tree matches, its not super critical I think. Its more just a convenience and double checking my work